journal/journal-app.js

```diff
@@ -32,69 +31,121 @@ function addJournalNew(journalContainer, existing) {
+    const journalElementTest = document.createElement('div');
+    journalElementTest.classList.add('journal-widget');
+    // journalElementTest.textContent = 'Insert Title';
+    const journalWidgetTitle = document.createElement('span')
+    journalWidgetTitle.classList.add('journal-widget-title')
+    journalWidgetTitle.textContent = 'Insert Title';
+    journalElementTest.append(journalWidgetTitle);
+    journalElementTest.setAttribute('widget-id', journalID);
+
+    // Create a container for the buttons
+    var buttonContainer = document.createElement('div');
+    buttonContainer.style.position = 'absolute';
+    buttonContainer.style.top = '5px';
+    buttonContainer.style.right = '5px';
+
+    // Create the 'Edit' button
+    var editButton = document.createElement('button');
+    editButton.innerText = 'Edit';
+    editButton.style.fontSize = '12px';
+    editButton.style.padding = '5px 10px';
+    editButton.style.marginRight = '5px';  // Space between buttons
+
+    // Create the 'Delete' button
+    var deleteButton = document.createElement('button');
+    deleteButton.innerText = 'Delete';
+    deleteButton.style.fontSize = '12px';
+    deleteButton.style.padding = '5px 10px';
+
+    // Append buttons to the container
+    buttonContainer.appendChild(editButton);
+    buttonContainer.appendChild(deleteButton);
+
+    // Ensure the parent div has relative positioning
+    journalElementTest.style.position = 'relative';
+
+    // Append the button container to the div
+    journalElementTest.append(buttonContainer);
+
+    // Function to open the journal modal
+    function openJournalModal(event) {
+        let overlay = document.getElementById('overlay');
+        overlay.style.display = 'block';
+        
+        // Use closest to find the parent element with the journal-widget class
+        const widget = event.target.closest('.journal-widget');
+        const currentJournalID = widget.getAttribute('widget-id'); // Correctly get the widget-id
         
-            journalElementTest.addEventListener('dblclick', function(event) {
-                document.getElementById('overlay').style.display = 'block';
-                const currentJournalID = event.target.getAttribute('widget-id');
-                const journalEntries = document.querySelectorAll('.journal-entry');
-                hideOtherJournalEntries(journalEntries, currentJournalID);
-                document.getElementById('modal').style.display = 'block';
-            });
-            
-            journalContainer.append(journalElementTest);
-            
-            // Save new journal to localStorage
-            journalList.push(journalTemplate);
-            saveNotes(journalList);
+        const journalEntries = document.querySelectorAll('.journal-entry');
+        hideOtherJournalEntries(journalEntries, currentJournalID);
+        let modal = document.getElementById('modal');
+        modal.style.display = 'block';
```

Comment:
Should this function be outside of the `createJournalWidget` function?

Reply:
moved it outside of the createJournalWidget function